### ISSUE TITLE
[BUG] Missing timezone for some photos

### ISSUE THREAD

**Issue by uhthomas** [immich-app/immich#2953]
### The bug

A user in Discord has found the attached image to show no time zone when uploaded with Immich. A quick inspection with `exiftool` reveals:

```sh
ExifTool Version Number         : 12.60
File Name                       : image00001.jpeg
Directory                       : .
File Size                       : 14 kB
File Modification Date/Time     : 2023:06:25 17:16:13+01:00
File Access Date/Time           : 2023:06:25 17:16:13+01:00
File Inode Change Date/Time     : 2023:06:25 17:16:15+01:00
File Permissions                : -rw-r--r--
File Type                       : JPEG
File Type Extension             : jpg
MIME Type                       : image/jpeg
JFIF Version                    : 1.01
Exif Byte Order                 : Big-endian (Motorola, MM)
Orientation                     : Horizontal (normal)
X Resolution                    : 72
Y Resolution                    : 72
Resolution Unit                 : inches
Modify Date                     : 2023:06:24 19:27:33
Exif Version                    : 0221
Date/Time Original              : 2023:06:24 19:27:33
Create Date                     : 2023:06:24 19:27:33
Offset Time                     : +05:30
Offset Time Original            : +05:30
Offset Time Digitized           : +05:30
Components Configuration        : Y, Cb, Cr, -
Sub Sec Time                    : 000
Sub Sec Time Original           : 000
Sub Sec Time Digitized          : 000
Flashpix Version                : 0100
Color Space                     : Uncalibrated
Exif Image Width                : 135
Exif Image Height               : 574
Scene Capture Type              : Standard
GPS Latitude Ref                : North
GPS Longitude Ref               : East
GPS Time Stamp                  : 05:27:30
GPS Date Stamp                  : 2023:06:25
GPS Horizontal Positioning Error: 1 m
Current IPTC Digest             : 271cf5277073fe39030fc3e607e77e93
Coded Character Set             : UTF8
Application Record Version      : 2
Digital Creation Time           : 19:27:33
Digital Creation Date           : 2023:06:24
Date Created                    : 2023:06:24
Time Created                    : 19:27:33+05:30
IPTC Digest                     : 271cf5277073fe39030fc3e607e77e93
Profile CMM Type                : Apple Computer Inc.
Profile Version                 : 4.0.0
Profile Class                   : Display Device Profile
Color Space Data                : RGB
Profile Connection Space        : XYZ
Profile Date Time               : 2022:01:01 00:00:00
Profile File Signature          : acsp
Primary Platform                : Apple Computer Inc.
CMM Flags                       : Not Embedded, Independent
Device Manufacturer             : Apple Computer Inc.
Device Model                    :
Device Attributes               : Reflective, Glossy, Positive, Color
Rendering Intent                : Perceptual
Connection Space Illuminant     : 0.9642 1 0.82491
Profile Creator                 : Apple Computer Inc.
Profile ID                      : ecfda38e388547c36db4bd4f7ada182f
Profile Description             : Display P3
Profile Copyright               : Copyright Apple Inc., 2022
Media White Point               : 0.96419 1 0.82489
Red Matrix Column               : 0.51512 0.2412 -0.00105
Green Matrix Column             : 0.29198 0.69225 0.04189
Blue Matrix Column              : 0.1571 0.06657 0.78407
Red Tone Reproduction Curve     : (Binary data 32 bytes, use -b option to extract)
Chromatic Adaptation            : 1.04788 0.02292 -0.0502 0.02959 0.99048 -0.01706 -0.00923 0.01508 0.75168
Blue Tone Reproduction Curve    : (Binary data 32 bytes, use -b option to extract)
Green Tone Reproduction Curve   : (Binary data 32 bytes, use -b option to extract)
Image Width                     : 135
Image Height                    : 574
Encoding Process                : Baseline DCT, Huffman coding
Bits Per Sample                 : 8
Color Components                : 3
Y Cb Cr Sub Sampling            : YCbCr4:2:0 (2 2)
Image Size                      : 135x574
Megapixels                      : 0.077
Create Date                     : 2023:06:24 19:27:33.000+05:30
Date/Time Original              : 2023:06:24 19:27:33.000+05:30
Modify Date                     : 2023:06:24 19:27:33.000+05:30
GPS Date/Time                   : 2023:06:25 05:27:30Z
GPS Latitude                    : 23 deg 0' 54.77" N
GPS Longitude                   : 72 deg 35' 33.51" E
Date/Time Created               : 2023:06:24 19:27:33+05:30
Digital Creation Date/Time      : 2023:06:24 19:27:33
GPS Position                    : 23 deg 0' 54.77" N, 72 deg 35' 33.51" E
```

Note, there are two entries for the `Create Date` tag. The first one has no timezone, whereas the second one _does_ have a timezone. We should explore whether #2908 resolves this and investigate further as to why this happens and how we can prevent it from happening in future.

![image00001](https://github.com/immich-app/immich/assets/9749173/5e98889f-df8b-428f-b117-75f1df546824)

### The OS that Immich Server is running on

N/A

### Version of Immich Server

v1.63.2

### Version of Immich Mobile App

N/A

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
N/A
```

### Your .env content

```Shell
N/A
```

### Reproduction steps

```bash
N/A
```

**Comment by uhthomas** [comment]
FWIW this is resolved by #2908.

![image](https://github.com/immich-app/immich/assets/9749173/6ed16bdb-cf1b-461f-a2d7-691defa407e6)

**Comment by AngelaDMerkel** [comment]
I am commenting because this issue is still open and I have a similar problem. I am located in GMT+2 and have a combination of images from both my phone and a DSLR. The phone images are represented accurately in local time whereas the DSLR photos are two hours ahead from the moment they were captured. Exif on one of the DSLR images is as follows: 

```
ExifTool Version Number         : 12.60
File Name                       : 20230814.1431 — RICOH GR III-Enhanced-RD.TIF
Directory                       : /Users/duffy/Documents
File Size                       : 144 MB
File Modification Date/Time     : 2023:08:15 17:46:35+02:00
File Access Date/Time           : 2023:08:16 13:23:41+02:00
File Inode Change Date/Time     : 2023:08:16 13:23:39+02:00
File Permissions                : -rw-r--r--
File Type                       : TIFF
File Type Extension             : tif
MIME Type                       : image/tiff
Exif Byte Order                 : Little-endian (Intel, II)
Subfile Type                    : Full-resolution image
Image Width                     : 6000
Image Height                    : 4000
Bits Per Sample                 : 16 16 16
Compression                     : Uncompressed
Photometric Interpretation      : RGB
Make                            : RICOH IMAGING COMPANY, LTD.
Camera Model Name               : RICOH GR III
Strip Offsets                   : 30796
Samples Per Pixel               : 3
Rows Per Strip                  : 4000
Strip Byte Counts               : 144000000
X Resolution                    : 240
Y Resolution                    : 240
Planar Configuration            : Chunky
Resolution Unit                 : inches
Software                        : Adobe Photoshop Lightroom Classic 12.4 (Macintosh)
Modify Date                     : 2023:08:15 17:46:35
XMP Toolkit                     : Adobe XMP Core 7.0-c000 1.000000, 0000/00/00-00:00:00
Creator Tool                    : Adobe Photoshop Lightroom Classic 12.4 (Macintosh)
Rating                          : 1
Metadata Date                   : 2023:08:15 17:46:35+02:00
Label                           : Green
Document ID                     : xmp.did:7fd20eb1-e096-44f3-9e72-57cf3bfd679e
Preserved File Name             : _R015971.DNG
Original Document ID            : 9EE46199D705502EE41A02D5F38A864A
Instance ID                     : xmp.iid:7fd20eb1-e096-44f3-9e72-57cf3bfd679e
Format                          : image/tiff
Enhance Details Already Applied : True
Enhance Details Version         : 218300416
Distortion Correction Already Applied: True
Lateral Chromatic Aberration Correction Already Applied: True
Vignette Correction Already Applied: True
Raw File Name                   : 20230814.1431 — RICOH GR III-Enhanced-RD.dng
Version                         : 15.4
Process Version                 : 15.4
White Balance                   : Daylight
Color Temperature               : 5500
Tint                            : +10
Exposure 2012                   : +1.30
Contrast 2012                   : +7
Highlights 2012                 : -100
Shadows 2012                    : +46
Whites 2012                     : +12
Blacks 2012                     : -25
Texture                         : 0
Clarity 2012                    : 0
Dehaze                          : 0
Vibrance                        : +14
Parametric Shadows              : 0
Parametric Darks                : 0
Parametric Lights               : 0
Parametric Highlights           : 0
Parametric Shadow Split         : 25
Parametric Midtone Split        : 50
Parametric Highlight Split      : 75
Sharpen Radius                  : +1.0
Sharpen Detail                  : 25
Sharpen Edge Masking            : 0
Luminance Smoothing             : 0
Color Noise Reduction           : 25
Color Noise Reduction Detail    : 50
Color Noise Reduction Smoothness: 50
Hue Adjustment Red              : 0
Hue Adjustment Orange           : 0
Hue Adjustment Yellow           : 0
Hue Adjustment Green            : 0
Hue Adjustment Aqua             : 0
Hue Adjustment Blue             : 0
Hue Adjustment Purple           : 0
Hue Adjustment Magenta          : 0
Saturation Adjustment Red       : 0
Saturation Adjustment Orange    : 0
Saturation Adjustment Yellow    : 0
Saturation Adjustment Green     : 0
Saturation Adjustment Aqua      : 0
Saturation Adjustment Blue      : 0
Saturation Adjustment Purple    : 0
Saturation Adjustment Magenta   : 0
Luminance Adjustment Red        : 0
Luminance Adjustment Orange     : 0
Luminance Adjustment Yellow     : 0
Luminance Adjustment Green      : 0
Luminance Adjustment Aqua       : 0
Luminance Adjustment Blue       : 0
Luminance Adjustment Purple     : 0
Luminance Adjustment Magenta    : 0
Split Toning Shadow Hue         : 0
Split Toning Shadow Saturation  : 0
Split Toning Highlight Hue      : 0
Split Toning Highlight Saturation: 0
Split Toning Balance            : 0
Color Grade Midtone Hue         : 0
Color Grade Midtone Sat         : 0
Color Grade Shadow Lum          : 0
Color Grade Midtone Lum         : 0
Color Grade Highlight Lum       : 0
Color Grade Blending            : 50
Color Grade Global Hue          : 0
Color Grade Global Sat          : 0
Color Grade Global Lum          : 0
Auto Lateral CA                 : 1
Lens Profile Enable             : 1
Lens Manual Distortion Amount   : 0
Vignette Amount                 : 0
Defringe Purple Amount          : 0
Defringe Purple Hue Lo          : 30
Defringe Purple Hue Hi          : 70
Defringe Green Amount           : 0
Defringe Green Hue Lo           : 40
Defringe Green Hue Hi           : 60
Perspective Upright             : Auto
Perspective Vertical            : 0
Perspective Horizontal          : 0
Perspective Rotate              : 0.0
Perspective Aspect              : 0
Perspective Scale               : 100
Perspective X                   : 0.00
Perspective Y                   : 0.00
Grain Amount                    : 0
Post Crop Vignette Amount       : 0
Shadow Tint                     : 0
Red Hue                         : 0
Red Saturation                  : 0
Green Hue                       : 0
Green Saturation                : 0
Blue Hue                        : 0
Blue Saturation                 : 0
Override Look Vignette          : False
Tone Curve Name 2012            : Linear
Camera Profile                  : Adobe Standard
Camera Profile Digest           : 5AA331AFE2DA99780D45C89F4543C3FB
Lens Profile Setup              : LensDefaults
Lens Profile Name               : Adobe (Ricoh GR III 18.3mm f/2.8)
Lens Profile Filename           : RICOH  (RICOH GR III) - RAW.lcp
Lens Profile Digest             : 8819B1B0C61F561FEE26EE27F37ADB36
Lens Profile Is Embedded        : False
Lens Profile Distortion Scale   : 100
Lens Profile Vignetting Scale   : 100
Upright Version                 : 151388160
Upright Center Mode             : 0
Upright Center Norm X           : 0.500337
Upright Center Norm Y           : 0.500562
Upright Focal Mode              : 0
Upright Focal Length 35mm       : 28
Upright Preview                 : False
Upright Dependent Digest        : C38DD264AB15290A313A4D6EED70918B
Upright Transform Count         : 6
Upright Transform 0             : 1.000000000,0.000000000,0.000000000,0.000000000,1.000000000,0.000000000,0.000000000,0.000000000,1.000000000
Upright Transform 1             : 1.028613489,0.013256628,-0.020935058,-0.029827414,1.028613489,0.000606962,0.000000000,0.000000000,1.000000000
Upright Transform 2             : 1.028613489,0.013256628,-0.020935058,-0.029827414,1.028613489,0.000606962,0.000000000,0.000000000,1.000000000
Upright Transform 3             : 1.028616405,0.013256666,-0.020936535,-0.029827499,1.028616406,0.000605546,0.000000000,0.000000000,1.000000000
Upright Transform 4             : 1.028613489,0.013256628,-0.020935058,-0.029827414,1.028613489,0.000606962,0.000000000,0.000000000,1.000000000
Upright Transform 5             : 1.000000000,0.000000000,0.000000000,0.000000000,1.000000000,0.000000000,0.000000000,0.000000000,1.000000000
Has Settings                    : True
Crop Top                        : 0
Crop Left                       : 0
Crop Bottom                     : 1
Crop Right                      : 1
Crop Angle                      : 0
Crop Constrain To Warp          : 0
Has Crop                        : False
Already Applied                 : True
History Action                  : derived, saved, derived, saved, derived, saved
History Parameters              : saved to new location, saved to new location, converted from image/dng to image/tiff, saved to new location
History Instance ID             : xmp.iid:6aa174ac-9747-4aa0-9342-b59ca69732b3, xmp.iid:b348d108-c047-467e-9a86-70a57667c5b5, xmp.iid:7fd20eb1-e096-44f3-9e72-57cf3bfd679e
History When                    : 2023:08:15 13:47:52+02:00, 2023:08:15 17:38:37+02:00, 2023:08:15 17:46:35+02:00
History Software Agent          : Adobe Photoshop Lightroom Classic 12.4 (Macintosh), Adobe Photoshop Lightroom Classic 12.4 (Macintosh), Adobe Photoshop Lightroom Classic 12.4 (Macintosh)
History Changed                 : /, /, /
Derived From Instance ID        : xmp.iid:b348d108-c047-467e-9a86-70a57667c5b5
Derived From Document ID        : xmp.did:b348d108-c047-467e-9a86-70a57667c5b5
Derived From Original Document ID: 9EE46199D705502EE41A02D5F38A864A
Tone Curve PV2012               : 0, 0, 255, 255
Tone Curve PV2012 Red           : 0, 0, 255, 255
Tone Curve PV2012 Green         : 0, 0, 255, 255
Tone Curve PV2012 Blue          : 0, 0, 255, 255
Look Name                       : Adobe Color
Look Amount                     : 1
Look UUID                       : B952C231111CD8E0ECCF14B86BAA7077
Look Supports Amount            : false
Look Supports Monochrome        : false
Look Supports Output Referred   : false
Look Group                      : Profiles
Look Parameters Version         : 15.4
Look Parameters Process Version : 15.4
Look Parameters Convert To Grayscale: False
Look Parameters Camera Profile  : Adobe Standard
Look Parameters Look Table      : E1095149FDB39D7A057BAB208837E2E1
Look Parameters Tone Curve PV2012: 0, 0, 22, 16, 40, 35, 127, 127, 224, 230, 240, 246, 255, 255
Look Parameters Tone Curve PV2012 Red: 0, 0, 255, 255
Look Parameters Tone Curve PV2012 Green: 0, 0, 255, 255
Look Parameters Tone Curve PV2012 Blue: 0, 0, 255, 255
Current IPTC Digest             : 77a566f72ee57f250ae038ed6d31873d
Coded Character Set             : UTF8
Application Record Version      : 4
Date Created                    : 2023:08:14
Time Created                    : 14:31:36
Digital Creation Date           : 2023:08:14
Digital Creation Time           : 14:31:36
Photoshop Thumbnail             : (Binary data 15341 bytes, use -b option to extract)
IPTC Digest                     : 77a566f72ee57f250ae038ed6d31873d
Exposure Time                   : 1/1000
F Number                        : 5.6
Exposure Program                : Aperture-priority AE
ISO                             : 100
Sensitivity Type                : Standard Output Sensitivity
Standard Output Sensitivity     : 100
Exif Version                    : 0231
Date/Time Original              : 2023:08:14 14:31:36
Create Date                     : 2023:08:14 14:31:36
Offset Time                     : +02:00
Shutter Speed Value             : 1/1000
Aperture Value                  : 5.6
Exposure Compensation           : 0
Metering Mode                   : Multi-segment
Flash                           : Off, Did not fire
Focal Length                    : 18.3 mm
Color Space                     : Uncalibrated
Sensing Method                  : One-chip color area
Custom Rendered                 : Normal
Exposure Mode                   : Auto
Focal Length In 35mm Format     : 28 mm
Scene Capture Type              : Standard
Contrast                        : High
Saturation                      : Normal
Sharpness                       : Hard
Subject Distance Range          : Distant
Profile CMM Type                : Apple Computer Inc.
Profile Version                 : 4.0.0
Profile Class                   : Display Device Profile
Color Space Data                : RGB
Profile Connection Space        : XYZ
Profile Date Time               : 2022:01:01 00:00:00
Profile File Signature          : acsp
Primary Platform                : Apple Computer Inc.
CMM Flags                       : Not Embedded, Independent
Device Manufacturer             : Apple Computer Inc.
Device Model                    : 
Device Attributes               : Reflective, Glossy, Positive, Color
Rendering Intent                : Perceptual
Connection Space Illuminant     : 0.9642 1 0.82491
Profile Creator                 : Apple Computer Inc.
Profile ID                      : ecfda38e388547c36db4bd4f7ada182f
Profile Description             : Display P3
Profile Copyright               : Copyright Apple Inc., 2022
Media White Point               : 0.96419 1 0.82489
Red Matrix Column               : 0.51512 0.2412 -0.00105
Green Matrix Column             : 0.29198 0.69225 0.04189
Blue Matrix Column              : 0.1571 0.06657 0.78407
Red Tone Reproduction Curve     : (Binary data 32 bytes, use -b option to extract)
Chromatic Adaptation            : 1.04788 0.02292 -0.0502 0.02959 0.99048 -0.01706 -0.00923 0.01508 0.75168
Blue Tone Reproduction Curve    : (Binary data 32 bytes, use -b option to extract)
Green Tone Reproduction Curve   : (Binary data 32 bytes, use -b option to extract)
Aperture                        : 5.6
Image Size                      : 6000x4000
Megapixels                      : 24.0
Scale Factor To 35 mm Equivalent: 1.5
Shutter Speed                   : 1/1000
Modify Date                     : 2023:08:15 17:46:35+02:00
Date/Time Created               : 2023:08:14 14:31:36
Digital Creation Date/Time      : 2023:08:14 14:31:36
Circle Of Confusion             : 0.020 mm
Field Of View                   : 65.5 deg
Focal Length                    : 18.3 mm (35 mm equivalent: 28.0 mm)
Hyperfocal Distance             : 3.05 m
Light Value                     : 14.9
```
<img width="396" alt="Screenshot 2023-08-16 at 13 32 28" src="https://github.com/immich-app/immich/assets/63363428/73c4ee8f-ce33-49cf-a155-135e8dec9a6a">

In the attached image from the webview, you can see that the image is correctly identified as being GMT+2 but the taking time is also shifted two hours forward. The correct time should be `14:31 GMT+02:00` whereas in the web view it is `16:31 GMT+02:00`. These images are directly exported from lightroom which reflects their taking times accurately. Google Photos and my local system also correctly identify taking time.

**Comment by jrasm91** [comment]
You probably should set TZ on the container. If the image has no timeline, it falls back to the container timezone. If the container doesn't have a timezone it will shine UTC+0

**Comment by AngelaDMerkel** [comment]
The container specifies the offset as +2 which indicates that this is UTC+02:00. The thing is, this is an image exported directly from Lightroom and represents a pretty standard formatting for what I imagine are many users images. Having Immich properly manage the most popular raw processing software is probably a smart idea.

**Comment by jrasm91** [comment]
This doesn't have anything to do with lightroom. The original image does not have any timezone information in it:

```
Date/Time Original              : 2023:08:14 14:31:36
Create Date                     : 2023:08:14 14:31:36
```

No timezone = fallback to the timezone of the machine when metadata extraction happens. The "machine" is the immich-microservices docker container, which does not have a timezone by default, so the only thing that can be done is assume this picture was taken at `2023:08:14 14:31:36+0`

**Comment by AngelaDMerkel** [comment]
This does have to do with Lightroom. The way Lightroom is encoding the timezone information with with the `Offset Time` field.

```
Date/Time Original              : 2023:08:14 14:31:36
Create Date                     : 2023:08:14 14:31:36
Offset Time                     : +02:00
```

+02:00 is to specify it is CEST or GMT+02:00. Immich is simply not properly interpreting that data. Of the utilities I use, Immich is the only one which improperly identifies the timezone information.

**Comment by uhthomas** [comment]
We're looking to release something which should resolve this, but blocked on an upstream package. I think we'll pin to a branch instead of a formal release to move this along.

**Comment by AngelaDMerkel** [comment]
@jrasm91 Apologies if I was unclear. I could have been more precise.

**Comment by AngelaDMerkel** [comment]
I have been testing this more to see any other edge cases which might occur, and there are several, including in cases where the camera time is incorrect and then shifted forward some hours into a different timezone. The metadata for this file is: 

```
Modify Date                     : 2023:08:17 21:03:21+02:00
Create Date                     : 2019:02:12 22:01:27-05:00
Metadata Date                   : 2023:08:17 21:03:21+02:00
```
Immich is using `create date` to lift timezone information, but that is simply not correct and doesn't reflect the actual time of the image.

**Comment by bo0tzz** [comment]
Fixed by #2908
